### ISSUE TITLE
How to Adjust a River Plot to Reduce Label Overlap?

### ISSUE THREAD

**Issue by km-2021** [ChristofferFlensburg/superFreq#128]
I am working with river plots and I am wondering if there is a way to increase the width of the plot figure and reduce the overlap of labels, perhaps by making the text for the variants smaller or using other methods. Any suggestions would be greatly appreciated!

Thank you in advance for your help.

**Comment by ChristofferFlensburg** [comment]
First, river plots, as most superFreq plots, are intended for data exploration, not as publication level figure panels, which is why de-clutter in plots is at low priority throughout. So if you're going for publication level viz, I'd start by reconsidering what you want to show, and what is the best way to present your results in a compact way. It might not be a river plot. I also want to remind that river plots by default contains germline information about human samples, so it's important to be aware of consent and how you are allowed to share that kind of data.

If you do want to do de-clutter rivers, easiest way is to use a vector graphics editor like illustrator or infinity.

There are also other packages entirely dedicated to this kind of viz, fish plots for example is a package that wrote a full paper on these plots alone. But it'd be some work to export the data to fit to their format.

Last, if you really want to do it through superFreq, then you'd have to (make a copy and) edit the plotting function. I think you should be able to find the relevant function, let me know if you don't!

Sorry I can't help more.

**Comment by km-2021** [comment]
Thank you for your reply and your assistance. If I specify the normal sample’s .bam and .vcf files at the beginning of the input data and run the analysis, will the germline information still be included in the results?

________________________________
差出人: Christoffer Flensburg ***@***.***>
送信日時: 2024年12月18日 8:18
宛先: ChristofferFlensburg/superFreq ***@***.***>
CC: 真砂 勝泰 ***@***.***>; Author ***@***.***>
件名: Re: [ChristofferFlensburg/superFreq] How to Adjust a River Plot to Reduce Label Overlap? (Issue #128)


First, river plots, as most superFreq plots, are intended for data exploration, not as publication level figure panels, which is why de-clutter in plots is at low priority throughout. So if you're going for publication level viz, I'd start by reconsidering what you want to show, and what is the best way to present your results in a compact way. It might not be a river plot. I also want to remind that river plots by default contains germline information about human samples, so it's important to be aware of consent and how you are allowed to share that kind of data.

If you do want to do de-clutter rivers, easiest way is to use a vector graphics editor like illustrator or infinity.

There are also other packages entirely dedicated to this kind of viz, fish plots for example is a package that wrote a full paper on these plots alone. But it'd be some work to export the data to fit to their format.

Last, if you really want to do it through superFreq, then you'd have to (make a copy and) edit the plotting function. I think you should be able to find the relevant function, let me know if you don't!

Sorry I can't help more.

**Comment by ChristofferFlensburg** [comment]
Yes, rare germline variants are shown with or without matched normal.

We've had data sets where we weren't supposed to even look at germline variants ourselves (according to patient concent), so I added a setting to `superFreq()` that tries to hide it from output. The documentation actually explains it pretty well:
```
rareGermline: logical. Display rare germline variants. This is a
          by-product of the clonal tracking and can yield important
          information about the sample, but due to ethics this is not
          always desired. This options allows a user to run superFreq
          while avoiding germline information. The user should keep in
          mind that superFreq, like most open source software, comes
          with no warranty and there is always a risk of bugs producing
          unintended behaviour. Also note that for individuals without
          matched normal it can be difficult to separate cancer
          variants present in most cells in all samples from germline
          variants. So rare germline variants may be misidentified as
          somatic variants without a matched normal.
```
In the case of the river, I think it just removes the black variants associated with the germline clone, so if that's all you need, you could get the same effect easier by editing in illustrator.
